### ISSUE TITLE
[Bug] Error with arrays and ValidatorFunctions

### ISSUE THREAD

**Issue by BastiOfBerlin** [T-Systems-MMS/openapi-typescript-angular-generator#2]
Hi,

I've got a 'People' model which is simply an array of persons:
```
People:
  type: array
  items:
    $ref: "#/components/schemas/Person"
```

This generates `export const PeopleValidators: {[K in keyof People]: Array<ValidatorFn>} = { };`, which results in the following transpilation error:

`ERROR in src/app/core/generated/api/model/people.ts(30,14): error TS2322: Type '{}' is not assignable to type '{ [x: number]: ValidatorFn[]; length: ValidatorFn[]; toString: ValidatorFn[]; toLocaleString: ValidatorFn[]; pop: ValidatorFn[]; push: ValidatorFn[]; concat: ValidatorFn[]; join: ValidatorFn[]; reverse: ValidatorFn[]; ... 21 more ...; includes: ValidatorFn[]; }'.
  Property 'length' is missing in type '{}'.`

I don't know if simply assigning `null` instead of empty object conflicts with the form-control-factory, so I didn't try..

I assume that this error will occur with every object that doesn't have own properties.

**Comment by malaupa** [comment]
Hopefully, we will have a look at this before Xmas. This should be fixed.

**Comment by malaupa** [comment]
@BastiOfBerlin could you please test, if the fix works for your problem?

**Comment by BastiOfBerlin** [comment]
Interestingly, the issue seemed to have vanished by using the newer generator as the problematic people.ts isn't generated at all. Instead, the API service implements the return type directly as Observable<Array<Person>>.

But to answer your question: I fiddled around and cherry-picked your fix into the old 1.2.0 release and YES, making the validator property optional worked just fine!

**Comment by cogidoo** [comment]
This bug is indeed solved with 3.3.4 of openapitools-generator (see [#1296](https://github.com/OpenAPITools/openapi-generator/pull/1296)).

**Comment by malaupa** [comment]
@cogidoo regardless the fix of the generator, we should merge. I think it should be possible to create a factory only with a fewer count of properties and their validators. Does the optional flag break something?
